src/Ast.hs

```diff
@@ -22,12 +22,17 @@ type OperatorId = String
 
 data Program = Program [Declaration]
 
-data Declaration = Function VarId [Bind] Expr
+data FnTyAnnotation = ReturnType Ty
+                    | CurriedType Ty
 
-data Bind = Bind [VarId] Ty
+data Declaration = Function VarId [Bind] Expr FnTyAnnotation
+
+data Bind = Bind VarId (Maybe Ty)
```

Comment:
Make VarId be optional, allow something like `foo () () ()`, `foo _ _ _ `

Reply:
@j-hui , making it optional means losing whether it was a () or a _. I am just making the `VarId` whatever is matched for now.

Reply:
Yeah this makes sense. I won't make this a blocker for this PR, but I think `Bind` should eventually resemble something like what is found in the pattern of a pattern-match arm.